Intermittent issue in testing line-based target function

On current `master`, I intermittently get this test error: 

```
======================================================================
FAIL: dipy.tracking.tests.test_utils.test_target_lb
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/Users/arokem/source/dipy/dipy/tracking/tests/test_utils.py", line 237, in test_target_lb
    _target(target_line_based, streamlines, (1, 1, 1), (2, 1, 1), False)
  File "/Users/arokem/source/dipy/dipy/tracking/tests/test_utils.py", line 276, in _target
    assert_equal(len(new), 1)
AssertionError: 2 != 1

----------------------------------------------------------------------
```

This probably has something to do with the values in the affine that is randomly set on L272, but I couldn't figure out what values cause this to happen.

--- a/dipy/tracking/tests/test_utils.py
+++ b/dipy/tracking/tests/test_utils.py
@@ -269,7 +269,11 @@ def _target(target_f, streamlines, voxel_both_true, voxel_one_true,
         assert_raises(ValueError, list, new)
 
     # Test smaller voxels
-    affine = np.random.random((4, 4)) - .5
+    random_array = np.array([[0.2862315, 0.44142904, 0.19837613, 0.422711],
+                             [0.7434331, 0.55335599, 0.18633461, 0.427429],
+                             [0.1465992, 0.84593179, 0.79033433, 0.576709],
+                             [0.6525159, 0.96735703, 0.69833409, 0.117800]])
+    affine = random_array - .5
     affine[3] = [0, 0, 0, 1]
     streamlines = list(move_streamlines(streamlines, affine))
     new = list(target_f(streamlines, mask, affine=affine))

--- a/dipy/tracking/tests/test_utils.py
+++ b/dipy/tracking/tests/test_utils.py
@@ -269,12 +269,10 @@ def _target(target_f, streamlines, voxel_both_true, voxel_one_true,
         assert_raises(ValueError, list, new)
 
     # Test smaller voxels
-    random_array = np.array([[0.2862315, 0.44142904, 0.19837613, 0.422711],
-                             [0.7434331, 0.55335599, 0.18633461, 0.427429],
-                             [0.1465992, 0.84593179, 0.79033433, 0.576709],
-                             [0.6525159, 0.96735703, 0.69833409, 0.117800]])
-    affine = random_array - .5
-    affine[3] = [0, 0, 0, 1]
+    affine = np.array([[.3, 0, 0, 0],
+                       [0, .2, 0, 0],
+                       [0, 0, .4, 0],
+                       [0, 0, 0, 1]])
     streamlines = list(move_streamlines(streamlines, affine))
     new = list(target_f(streamlines, mask, affine=affine))
     assert_equal(len(new), 1)

